cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -5024,6 +5024,41 @@ class TestBufferedParquetIO : public TestParquetIO<TestType> {
     ASSERT_OK_NO_THROW(writer->Close());
   }
 
+  void WriteBufferedTable(const std::shared_ptr<Array>& values,
```

Comment:
Can you please refactor this with `WriteBufferedFile` to avoid copy-pasting entire chunks of code?

Reply:
Hmmm I guess I use some different logic here, I'll try to unify them